arviz/stats/stats.py

```diff
@@ -1142,9 +1146,26 @@ def summary(
     else:
         if not 1 >= hdi_prob > 0:
             raise ValueError("The value of hdi_prob should be in the interval (0, 1]")
-    posterior = convert_to_dataset(data, group="posterior", **extra_args)
-    var_names = _var_names(var_names, posterior, filter_vars)
-    posterior = posterior if var_names is None else posterior[var_names]
+
+    if isinstance(data, InferenceData):
+        if group is None:
+            if not data.groups():
+                raise TypeError("InferenceData does not contain any groups")
+            if "posterior" in data:
+                dataset = data["posterior"]
+            elif "prior" in data:
```

Comment:
should we have `kind` default to `stats` in some cases? diagnostics are not really useful for prior samples

Reply:
We could, unless user uses mcmc to do the prior predictive sampling. (e.g. in Stan defining parameters and only priors --> this can lead to autocorrelation, but could be easier to do with complex models)

Reply:
Let's merge like this then